packages/ui/src/modal/TokenPrimitive.tsx

```diff
@@ -90,6 +98,34 @@ const TokenPrimitive: FC<Props> = ({
               </Text>
             )}
             {!!expires && <Text style="tiny">Expires {expires}</Text>}
+            {!expires && quantity && quantity > 1 ? (
+              <Flex
+                css={{
+                  p: '$1 ',
+                  background: '$neutralBgHover',
+                  borderRadius: 4,
+                  mr: 'auto',
+                }}
+              >
+                <Text style="tiny" color="base">
+                  {quantity} {quantity > 1 ? 'items' : 'item'}
+                </Text>
+              </Flex>
+            ) : null}
+            {!expires && !quantity && royaltiesBps ? (
+              <Text
+                style="body2"
+                color="subtle"
+                css={{ display: 'flex', gap: '$1' }}
+              >
+                Creator Royalties: {royaltyPercent}
+                <InfoTooltip
+                  side="right"
+                  width={200}
+                  content="A fee on every order that goes to the collection creator."
+                />
+              </Text>
+            ) : null}
           </Grid>
         </Flex>
         <Grid css={{ justifyItems: 'end', alignContent: 'start', rowGap: 4 }}>
```

Comment:
The price flickers every so often when increasing quantity. I think it happens when switching between different orders. I wonder if there is a way to fix this 🤔

Reply:
Fixed, this was due to quantity changing independently of the total price and vice versa. Added a piece of state to store the average unit price whenever calculating the total price, everything is now in sync and looking better.